src/Views/VPNPage.vala

```diff
@@ -107,9 +106,16 @@ public class Network.VPNPage : Network.Widgets.Page {
 
         show_all ();
 
+        vpn_list.row_activated.connect (row => {
+            if (((VPNMenuItem) row).state == State.CONNECTED_VPN) {
+                disconnect_vpn_cb ((VPNMenuItem) row);
+            } else {
+                connect_vpn_cb ((VPNMenuItem) row);
+            }
+        });
+
         vpn_list.row_selected.connect (row => {
-            remove_button.sensitive = true;
-            edit_connections_button.sensitive = true;
```

Comment:
I am of the opinion the `edit_connections_button` should also change sensitivity and remove the possibility of triggering `nm-connection-editor --type=vpn -s` . Instead of showing users 2 types of ugly dialogs show only one. The list only allows for addition, deletion and editing, which the main list does anyways.